### PR TITLE
Feature/checkbox

### PR DESCRIPTION
**Pull request by hadasfa**
## Go over this checklist before submitting your Pull Request

Description: Add support for a new checkbox component.

#### Basic
- [x] Used plop (npm run plop) to create a new component. This command will create a folder for your component and will update index.js file
- [x] PR has description
- [x] New component is functional and using React Hooks. 
- [x] Component defines PropTypes
#### Style
- [x] Styles are added to NewComponent.scss file inside of the NewComponent folder
- [x] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [x] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [x] Stories were added to /src/NewComponent/__stories__/NewComponent.stories.js file
- [x] Stories include sandbox story with knobs for each property
- [x] Stories include all flows of using the component
- [x] Component passed Accessibility Plugin checks
#### Tests
- [x] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

File: src/StoryBookComponents/StoryWrapper/StoryWrapper.jsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import React from "react";
 import "./StoryWrapper.scss";
+import cx from "classnames";
 
-const StoryWrapper = ({ children }) => {
-  return <div className="monday-style-story-wrapper">{children}</div>;
+const StoryWrapper = ({ componentClassName, children }) => {
```
Comment (orrgottlieb) [inline src/StoryBookComponents/StoryWrapper/StoryWrapper.jsx:5]: yay!!!

File: src/components/Checkbox/Checkbox.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  label: PropTypes.string,
+  onChange: PropTypes.func,
+  checked: PropTypes.bool,
+  defaultChecked: PropTypes.bool,
+  value: PropTypes.string,
+  disabled: PropTypes.bool,
+  name: PropTypes.string
+};
+
+Checkbox.defaultProps = {
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.js:69]: please add default props to all props

File: src/components/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  componentClassName: "",
+  label: "",
+  onChange: () => {},
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.js:72]: please change () => {} to noop from lodash
`import NOOP from "lodash/noop";`

File: src/components/Checkbox/Checkbox.scss [hunk 4]
```diff
@@ -0,0 +1,149 @@
+@import "../../styles/global-css-settings";
+@import "../../styles/themes.scss";
+@import "../../styles/typography.scss";
+
+$checkbox-size: 16px;
+
+@mixin hover-selected {
+  background-color: var(--primary-hover-color);
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.scss:8]: use theme-prop 
`@include theme-prop(background-color, primary-hover-color);`

File: src/components/Checkbox/Checkbox.scss [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    cursor: pointer;
+    visibility: visible;
+    display: flex;
+    justify-content: center;
+    flex-direction: column;
+    height: $checkbox-size;
+    width: $checkbox-size;
+    border: 1px solid var(--ui-border-color);
+    border-radius: $border-radius-extra-small;
+    transition: transform 50ms cubic-bezier(0, 0, 0.35, 1);
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.scss:43]: the transition - we have a sass constat for it

File: src/components/Checkbox/Checkbox.scss (hunk 4 above)
Comment (orrgottlieb) [reply src/components/Checkbox/Checkbox.scss:8]: please change for the entire file

File: src/components/Checkbox/Checkbox.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  defaultChecked,
+  value,
+  name
+}) => {
+  const checkboxClassNames = [`${BASE_CLASS_NAME}__checkbox`];
+  let overrideDefaultChecked = defaultChecked;
+
+  // If component did not receive default checked and checked props, choose default checked as
+  // default behavior (handle isChecked logic inside input) and set default value
+  if (overrideDefaultChecked == undefined && checked == undefined) {
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.js:25]: why "==" ? and not "==="

File: src/components/Checkbox/Checkbox.scss [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+.monday-style-checkbox {
+  position: relative;
+  display: flex;
+  align-items: center;
+  width: fit-content;
+  height: min-content;
+
+  &:active {
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.scss:23]: love it!

File: src/components/Checkbox/Checkbox.scss [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    @include font-input();
+    cursor: pointer;
+    margin-inline-start: $spacing-small;
+    color: var(--primary-text-color);
+  }
+
+  // Since it is not possible to change the design of the checkbox according to the requirements using css,
+  // we hide the checkbox and draw a new one instead.
+  // In order to allow accessibility, all operations will be performed on the hidden checkbox and be reflected
+  // in the new checkbox we drew.
```
Comment (orrgottlieb) [inline src/components/Checkbox/Checkbox.scss:80]: awesome comment!

File: src/components/Checkbox/__stories__/checkbox.stories.js [hunk 9]
```diff
@@ -0,0 +1,162 @@
+import React, { useState } from "react";
```
Comment (orrgottlieb) [inline src/components/Checkbox/__stories__/checkbox.stories.js:1]: very cool!

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

**Comment by orrgottlieb** [comment]
Awesome work!!!!

File: src/components/Checkbox/Checkbox.js (hunk 6 above)
Comment (hadasfa) [reply src/components/Checkbox/Checkbox.js:25]: Because it check both undefined and null

Comment (orrgottlieb) [reply src/components/Checkbox/Checkbox.js:25]: we prefer not to use the Abstract Equality Operator ("==") I think we can change it to be more explicit or use lodash isNil in my opinion it is clearer

File: src/components/Checkbox/Checkbox.scss (hunk 5 above)
Comment (hadasfa) [reply src/components/Checkbox/Checkbox.scss:43]: something like this - @include transition(transform 50ms cubic-bezier(0, 0, 0.35, 1);  ?

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**